### PR TITLE
Streamlining the Base and adding features

### PR DESCRIPTION
**Pull request by Pootisp**
Changes
Most of the common npcs now can shoot at long ranges
-REBELS-
--Now give you 45 smg ammo. removed the pistol and shotgun
-REBEL SHOTGUNNER-
--A dedicated rebel who uses shotgun. give you 8 buckshots
-ANTLION GUARD-
now able to summon antlions by default. cave variant too
-METROPOLICE-
Metropolice now has 3 flavours to choose from
Metro Police 
Metropolice (simple)- these guys are from the early levels. have 26 health. and flinch more
Metrocop Manhack Spammer (this guy is terrifying to fight against. recommend 1v5)
-COMBINE SOLDIERS-
Now have Infinite grenades and a longer range.
-HUNTER-CHOPPER-
Restored its massive 5600 health and added lights
-HECU GRUNTS-
Now come in 3 flavours. again and now able to use squads*
The run of the mill guys
The Cigar chewing asshat that leads the squad and shoot grenades
and shotgun dude

-HOUNDEYES-
Now they can uses squads*

NOTE
*You need to spawn the leader first before the rest
*for some reason you can only have 1 squad at a time. so dont spawn in grunts with houndeyes or else there will be errors!

### CODE REVIEW

File: garrysmod/lua/autorun/base_npcs.lua [hunk 1]
```diff
@@ -14,15 +14,15 @@ AddNPC( {
 	Name = "Alyx Vance",
 	Class = "npc_alyx",
 	Category = Category,
-	Weapons = { "weapon_alyxgun", "weapon_smg1", "weapon_shotgun" },
+	Weapons = { "weapon_alyxgun" },
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:17]: Pointless change. Do not remove any weapons from these lists.

File: garrysmod/lua/autorun/base_npcs.lua [hunk 2]
```diff
@@ -51,6 +51,7 @@ AddNPC( {
 } )
 
 -- Did you know that this MAN can shoot annabelle like he's been doing it his whole life?
+ -- ^ Bullshit. HE cant shot SHIT. except pistols strangely..... but man can he shoot pistols
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:54]: Why is this here?

File: garrysmod/lua/autorun/base_npcs.lua [hunk 3]
```diff
@@ -306,11 +316,29 @@ AddNPC( {
...
 	Category = Category,
-	Weapons = { "weapon_stunstick", "weapon_pistol", "weapon_smg1" },
-	SpawnFlags = SF_NPC_DROP_HEALTHKIT,
-	KeyValues = { SquadName = "overwatch" }
+	Weapons = { "weapon_pistol", "weapon_smg1" },
+	SpawnFlags = SERVER and bit.bor( SF_NPC_DROP_HEALTHKIT, SF_NPC_LONG_RANGE ) or nil,
+	KeyValues = { SquadName = "overwatch", manhacks = 2 }
 } )
 
+AddNPC( {
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:324]: How is this one different from a normal metrocop?

File: garrysmod/lua/autorun/base_npcs.lua [hunk 4]
```diff
@@ -306,11 +316,29 @@ AddNPC( {
...
+	KeyValues = { SquadName = "overwatch", manhacks = 1 }
+}, "Simple Metropolice" )
+
+AddNPC( {
+	Name = "Metro Police Manhack Spammer ",
+	Class = "npc_metropolice",
+	Category = Category,
+	Weapons = { "weapon_357" },
+	SpawnFlags = SERVER and bit.bor( SF_NPC_DROP_HEALTHKIT, SF_NPC_LONG_RANGE ) or nil,
+	KeyValues = { SquadName = "overwatch", manhacks = 9999 }
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:339]: No thank you.

File: garrysmod/lua/autorun/base_npcs.lua [hunk 5]
```diff
@@ -336,8 +364,9 @@ AddNPC( {
 	Class = "npc_combine_s",
 	Category = Category,
 	Model = "models/combine_soldier.mdl",
-	Weapons = { "weapon_smg1", "weapon_ar2" },
-	KeyValues = { SquadName = "overwatch", Numgrenades = 5 }
+	Weapons = { "weapon_smg1", "weapon_ar2", },
+	KeyValues = { SquadName = "overwatch", Numgrenades = 500 },
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:368]: Why increase the grenade count?

File: garrysmod/lua/autorun/base_npcs.lua [hunk 6]
```diff
@@ -420,9 +453,10 @@ AddNPC( {
 	Class = "npc_helicopter",
 	Category = Category,
 	Offset = 300,
-	Health = 600,
+	Health = 5600,
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:456]: Why was this changed?

File: garrysmod/lua/autorun/base_npcs.lua [hunk 7]
```diff
@@ -420,9 +453,10 @@ AddNPC( {
 	Class = "npc_helicopter",
 	Category = Category,
 	Offset = 300,
-	Health = 600,
+	Health = 5600,
 	KeyValues = { SquadName = "overwatch" },
-	NoDrop = true
+	NoDrop = true,
+	SpawnFlags = 262144
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:459]: Please at least add comments about what these magic numbers mean.

File: garrysmod/lua/autorun/base_npcs.lua [hunk 8]
```diff
@@ -483,27 +517,172 @@ if ( IsMounted( "hl1" ) || IsMounted( "hl1mp" ) ) then
...
-	AddNPC( { Name = "Snark", Class = "monster_snark", Category = Category, Offset = 6, NoDrop = true } )
-	AddNPC( { Name = "Zombie", Class = "monster_zombie", Category = Category } )
-	AddNPC( { Name = "Headcrab", Class = "monster_headcrab", Category = Category } )
-	AddNPC( { Name = "Controller", Class = "monster_alien_controller", Category = Category, NoDrop = true } )
-	AddNPC( { Name = "Security Officer", Class = "monster_barney", Category = Category } )
-
-	AddNPC( { Name = "Heavy Turret", Class = "monster_turret", Category = Category, Offset = 0, KeyValues = { orientation = 1 }, OnCeiling = true, SpawnFlags = 32 } )
-	AddNPC( { Name = "Mini Turret", Class = "monster_miniturret", Category = Category, Offset = 0, KeyValues = { orientation = 1 }, OnCeiling = true, SpawnFlags = 32 } )
-	AddNPC( { Name = "Sentry", Class = "monster_sentry", Category = Category, Offset = 0, OnFloor = true, SpawnFlags = 32 } )
+AddNPC( {
```
Comment (robotboy655) [inline garrysmod/lua/autorun/base_npcs.lua:520]: What was the point in changing this section? Please undo.

**Review by robotboy655 — CHANGES_REQUESTED** [review]

**Review by GitSparTV — CHANGES_REQUESTED** [review]
It seems, in general, what you want is to buff default NPCs stats.
I think this work is for Workshop. Make it as separate addon: done.
